### PR TITLE
FIX [`Gemma` / `CI`] Make sure our runners have access to the model

### PR DESCRIPTION
**Pull request by younesbelkada**
# What does this PR do?

As discussed offline cc @ydshieh @sanchit-gandhi @ArthurZucker

### CODE REVIEW

**Review by ydshieh — APPROVED** [review]
I think it's fine to just put token as a top module variable. Kind a bit strange to has a class attribute as `token`.

Thanks for the  quick fix.

**Review by ArthurZucker — COMMENTED** [review]
Should the token not work as long as it's set with `export HF_HUB_READ_TOKEN=$GIT_SECRET`

**Review by ArthurZucker — COMMENTED** [review]
Thanks @Wauplin , after offline talks, seems that we should have something like: 
```python 
def require_read_token(fn):
    token = os.getenv("HF_HUB_READ_TOKEN")

    @wraps(fn)
    def _inner(*args, **kwargs):
        with patch(huggingface_hub.utils._headers, "get_token", return_value=token):
            return fn(*args, **kwargs)

    return _inner
```
Users who want to test locally are usually logged in, so if they don't pass a token (`False`) it should still be alright

File: tests/models/gemma/test_modeling_flax_gemma.py [hunk 1]
```diff
@@ -209,12 +208,13 @@ class FlaxGemmaIntegrationTest(unittest.TestCase):
...
+    token = os.getenv("HF_HUB_READ_TOKEN", False)
 
     def setUp(self):
         self.model, self.params = FlaxGemmaForCausalLM.from_pretrained(
-            self.model_id, revision=self.revision, _do_init=False
+            self.model_id, revision=self.revision, _do_init=False, token=self.token
         )
-        self.tokenizer = AutoTokenizer.from_pretrained(self.model_id)
+        self.tokenizer = AutoTokenizer.from_pretrained(self.model_id, token=self.token)
         self.tokenizer.padding_side = "left"
```
Comment (ArthurZucker) [inline tests/models/gemma/test_modeling_flax_gemma.py:218]: this is no longer needed

File: tests/models/gemma/test_modeling_gemma.py [hunk 2]
```diff
@@ -659,7 +660,10 @@ def test_model_2b_flash_attn(self):
         ]
 
         model = AutoModelForCausalLM.from_pretrained(
-            model_id, low_cpu_mem_usage=True, torch_dtype=torch.bfloat16, attn_implementation="flash_attention_2"
+            model_id,
+            low_cpu_mem_usage=True,
+            torch_dtype=torch.bfloat16,
+            attn_implementation="flash_attention_2",
```
Comment (ArthurZucker) [inline tests/models/gemma/test_modeling_gemma.py:666]: to revert?

**Review by ArthurZucker — APPROVED** [review]
LGTM otherwise

**Pull request merged**